pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -972,6 +972,22 @@ private void sendFlowPermitsToBroker(ClientCnx cnx, int numMessages) {
             if (log.isDebugEnabled()) {
                 log.debug("[{}] [{}] Adding {} additional permits", topic, subscription, numMessages);
             }
+
+            cnx.ctx().writeAndFlush(Commands.newFlow(consumerId, numMessages))
+                    .addListener(writeFuture -> {
+                        String msg;
+                        if (!writeFuture.isSuccess()) {
+                            msg = String.format("Consumer {} failed to send {} permits to broker: {}"
+                                    , consumerId, numMessages, writeFuture.cause().getMessage());
+                            increaseAvailablePermits(cnx, numMessages, false);
```

Comment:
Can you add a test to verify this? Is it possible that when the flush fails, the connection would be invalid and new connection will be created.